### PR TITLE
Ensure scheme exists and handle undefined signing keys

### PR DESCRIPTION
**Pull request by tonyhb**
This PR ensures that undefined signing keys are handled correctly, and ensures that we always have a scheme (required for node's URL parsing).

### CODE REVIEW

File: src/handlers/default.ts [hunk 1]
```diff
@@ -171,7 +175,24 @@ export class InngestCommHandler {
...
-      const reqUrl = new URL(req.originalUrl, req.hostname);
+      const hostname = req.hostname || req.headers["host"];
+      if (!hostname) {
+        console.error("Unable to determine your site URL to host Inngest.");
+        return res.status(500).json("Unable to determine your site URL to host Inngest.");
+      }
+
+      let scheme = "";
+      if (hostname.indexOf("://") === -1) {
+        scheme = (process.env.NODE_ENV === "development" || !req.connection.encrypted) ? "http://" : "https://";
```
Comment (djfarrelly) [inline src/handlers/default.ts:186]: I couldn't find `req.connection` cited in Express and it's marked as [deprecated in the Node docs](https://nodejs.org/dist/latest-v16.x/docs/api/http.html#requestconnection) - Can we use `req.protocol` instead?

* https://expressjs.com/en/api.html#req.protocol
* https://nodejs.org/dist/latest-v16.x/docs/api/http.html#requestprotocol

File: src/handlers/default.ts [hunk 2]
```diff
@@ -171,7 +175,24 @@ export class InngestCommHandler {
 
   public createHandler(): any {
     return async (req: Request, res: Response) => {
-      const reqUrl = new URL(req.originalUrl, req.hostname);
+      const hostname = req.hostname || req.headers["host"];
+      if (!hostname) {
+        console.error("Unable to determine your site URL to host Inngest.");
+        return res.status(500).json("Unable to determine your site URL to host Inngest.");
```
Comment (djfarrelly) [inline src/handlers/default.ts:181]: "host Inngest" feels "serve the Inngest handler"?

**Review by djfarrelly — COMMENTED** [review]
Good catch on this issue @tonyhb - I left a couple comments, one that may be good to fix up before shipping. The other thought is that we could extract this into another function or only run this code within the `PUT` part of the switch statement as those other bits don't need it, but that change isn't necessary.

File: src/handlers/default.ts [hunk 3]
```diff
@@ -98,7 +98,7 @@ export class InngestCommHandler {
   private readonly inngestRegisterUrl: URL;
 
   protected readonly frameworkName: string = "default";
-  protected readonly signingKey: string;
+  protected readonly signingKey: string | undefined;
```
Comment (jpwilliams) [inline src/handlers/default.ts:101]: Is it valid to not provide a signing key? Do we not always need one?

Comment (tonyhb) [reply src/handlers/default.ts:101]: we do need one, though if you use `proces.env.INNGEST_SIGNING_KEY` or similar env variables it's still possible to be undefined, so we need to denote this and handle appropriately.

File: src/handlers/default.ts (hunk 1 above)
Comment (tonyhb) [reply src/handlers/default.ts]: Done, thanks!

**Comment by tonyhb** [comment]
@djfarrelly There's plan with @jpwilliams to create a single interface that each custom framework handler fulfils to abstract this code.  This will DRY up any URL creation and is going to be handled separately from this PR.

**Pull request merged**